modules/core/shared/src/main/scala/lucuma/core/model/package.scala

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core
+
+import cats.Eq
+import cats.Monoid
+import cats.syntax.all._
+import eu.timepit.refined.api.{ Refined, RefinedTypeOps, Validate }
+import eu.timepit.refined.api.Validate.Plain
+import eu.timepit.refined.boolean.Not
+import eu.timepit.refined.numeric.{ GreaterEqual, Interval, Less, NonNegative }
+import shapeless.Nat._0
```

Comment:
Can this be done without shapeless?

Reply:
I really don't know. I just copied it from the api. 😄 I can look into it a little. Maybe @swalker2m knows more?

Reply:
I'll try replacing it.

Reply:
I'm going to merge, and then see if I can fix this. I'd like to get my api PR in before there are more merge conflicts.

Reply:
> I'm going to merge, and then see if I can fix this. I'd like to get my api PR in before there are more merge conflicts.

:+1:

Reply:
It turns out this probably isn't easily possible. It is relying on the fact that `Not` and `Less` are case classes with companion objects with the implicit validation functions. `NonNegative` is a type alias and so cannot have a companion object. I think in order to try to do the validation ourselves, we'd end up using the `shapeless.Nat._0`. Or we'd have to implement the `Nat`s.

Reply:
Never mind, I was mostly asking in the context of the scala 3 port